lib/stack_master/cloudformation_interpolating_eruby.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'erubis'
+
+module StackMaster
+  #  This class is a modified version of `Erubis::Eruby`. It allows using
+  # `<%= %>` ERB expressions to interpolate values into a source string. We use
+  # this capability to enrich user data scripts with data and parameters pulled
+  # from the AWS CloudFormation service. The evaluation produces an array of
+  # objects ready for use in a CloudFormation `Fn::Join` intrinsic function.
+  class CloudFormationInterpolatingEruby < Erubis::Eruby
```

Comment:
Extract this class out of the `stack_master/sparkle_formation/template_file.rb` file.

Allow the functionality to be reused between the SparkleFormation and YAML ERB template compilers.